microprofile/opentracing-smallrye/src/main/java/org/wildfly/extension/microprofile/opentracing/smallrye/TracingDeploymentProcessor.java

```diff
@@ -0,0 +1,124 @@
+    addListeners(deploymentUnit);
+    addJaxRsIntegration(deploymentUnit);
+    addCDIExtension(deploymentUnit);
+  }
+
+  @Override
+  public void undeploy(DeploymentUnit deploymentUnit) {
+  }
+
+  private void addListeners(DeploymentUnit deploymentUnit) {
+    JBossWebMetaData jbossWebMetaData = getJBossWebMetaData(deploymentUnit);
+    if (null == jbossWebMetaData) {
+      // nothing to do here
+      return;
+    }
+
+    TracingLogger.ROOT_LOGGER.registeringTracerInitializer();
+
+    String serviceName = getServiceName(deploymentUnit);
+    ParamValueMetaData serviceNameContextParameter = new ParamValueMetaData();
+    serviceNameContextParameter.setParamName(TracerInitializer.SMALLRYE_OPENTRACING_SERVICE_NAME);
+    serviceNameContextParameter.setParamValue(serviceName);
+    addContextParameter(jbossWebMetaData, serviceNameContextParameter);
+
+    ListenerMetaData listenerMetaData = new ListenerMetaData();
+    listenerMetaData.setListenerClass(TracerInitializer.class.getName());
+
+    List<ListenerMetaData> listeners = jbossWebMetaData.getListeners();
+    if (null == listeners) {
+      listeners = new ArrayList<>();
+    }
+    listeners.add(listenerMetaData);
+    jbossWebMetaData.setListeners(listeners);
+  }
+
+  private void addJaxRsIntegration(DeploymentUnit deploymentUnit) {
+    JBossWebMetaData jbossWebMetaData = getJBossWebMetaData(deploymentUnit);
+    if (null == jbossWebMetaData) {
+      // nothing to do here
+      return;
+    }
+
+    TracingLogger.ROOT_LOGGER.registeringJaxRs();
+
+    ParamValueMetaData restEasyProvider = new ParamValueMetaData();
+    restEasyProvider.setParamName("resteasy.providers");
+    restEasyProvider.setParamValue(SmallRyeTracingDynamicFeature.class.getName());
+    addContextParameter(jbossWebMetaData, restEasyProvider);
+  }
+
+  private void addCDIExtension(DeploymentUnit deploymentUnit) {
+    TracingLogger.ROOT_LOGGER.registeringCDIExtension();
+
+    WeldPortableExtensions extensions = WeldPortableExtensions.getPortableExtensions(deploymentUnit);
+    extensions.registerExtensionInstance(new TracingCDIExtension(), deploymentUnit);
+  }
+
+  private JBossWebMetaData getJBossWebMetaData(DeploymentUnit deploymentUnit) {
+    WarMetaData warMetaData = deploymentUnit.getAttachment(WarMetaData.ATTACHMENT_KEY);
+    if (null == warMetaData) {
+      // not a web deployment, nothing to do here...
+      return null;
+    }
+
+    return warMetaData.getMergedJBossWebMetaData();
+  }
+
+  private void addContextParameter(JBossWebMetaData jbossWebMetaData, ParamValueMetaData restEasyProvider) {
+    List<ParamValueMetaData> contextParams = jbossWebMetaData.getContextParams();
+    if (null == contextParams) {
+      contextParams = new ArrayList<>();
+    }
+    contextParams.add(restEasyProvider);
+    jbossWebMetaData.setContextParams(contextParams);
+  }
+
+  private String getServiceName(DeploymentUnit deploymentUnit) {
+    String serviceName = System.getProperty("JAEGER_SERVICE_NAME");
```

Comment:
This should use `WildFlySecurityManager.getPropertyPrivileged()`. That said I'm not sure I understand this. Controlling a service name depending on a system property or environment variable seems risky.

Reply:
Would you mind enumerating the risks? I'm not sure I fully understand them.

The idea is to keep the user's expectation regarding this env var/sysprop: Jaeger's service name is the only required configuration entry for Jaeger to work and it's usually set via env var/sysprop. *Not* supporting this way of configuring would make WildFly an exception and will probably confuse users.

We'll add a way to control this via WildFly's management model, but we'll likely keep supporting the configuration via env var.

Reply:
My apologies. I was thinking of an MSC `ServiceName`. This should use the `WildFlySecurityManager` though so the deployment will work with a security manager installed.

Reply:
Alright, I just changed it. I believe all your review items are now addressed.